src/lib.rs

```diff
@@ -2291,11 +2291,31 @@ impl Build {
             "powerpc-unknown-netbsd" => Some("powerpc--netbsd"),
             "powerpc64-unknown-linux-gnu" => Some("powerpc-linux-gnu"),
             "powerpc64le-unknown-linux-gnu" => Some("powerpc64le-linux-gnu"),
-            "riscv32i-unknown-none-elf" => Some("riscv32-unknown-elf"),
-            "riscv32imac-unknown-none-elf" => Some("riscv32-unknown-elf"),
-            "riscv32imc-unknown-none-elf" => Some("riscv32-unknown-elf"),
-            "riscv64gc-unknown-none-elf" => Some("riscv64-unknown-elf"),
-            "riscv64imac-unknown-none-elf" => Some("riscv64-unknown-elf"),
+            "riscv32i-unknown-none-elf" => self.find_working_gnu_prefix(&[
+                "riscv32-unknown-elf",
+                "riscv64-unknown-elf",
```

Comment:
Out of curiosity, can this 64-bit compiler be used for a 32-bit target?

Reply:
It can be, and very frequently is. The sifive toolchains, for example, advertise "Our toolchain and emulator distributions have been carefully packaged to support both 32-bit & 64-bit ISAs." You switch what you're targeting with `-march=`, which `cc` already does.